contracts/src/v0.8/l2ep/dev/scroll/ScrollSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,236 @@
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  uint256 public constant override version = 1;
+
+  /// @dev L1 address
+  address private s_l1Sender;
+  /// @dev s_latestRoundId == 0 means this contract is uninitialized.
+  FeedState private s_feedState = FeedState({latestRoundId: 0, latestStatus: false, startedAt: 0, updatedAt: 0});
+  mapping(uint80 roundId => Round round) private s_rounds;
+
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  IL2ScrollMessenger private immutable s_l2CrossDomainMessenger;
+
+  /// @param l1SenderAddress Address of the L1 contract that is permissioned to call this contract
+  /// @param l2CrossDomainMessengerAddr Address of the L2CrossDomainMessenger contract
+  /// @param initialStatus The initial status of the feed
+  constructor(address l1SenderAddress, address l2CrossDomainMessengerAddr, bool initialStatus) {
+    _setL1Sender(l1SenderAddress);
+    s_l2CrossDomainMessenger = IL2ScrollMessenger(l2CrossDomainMessengerAddr);
+    uint64 timestamp = uint64(block.timestamp);
+
+    // Initialise roundId == 1 as the first round
+    _recordRound(1, initialStatus, timestamp);
+  }
+
+  /// @notice Check if a roundId is valid in this current contract state
+  /// @dev Mainly used for AggregatorV2V3Interface functions
+  /// @param roundId Round ID to check
+  function _isValidRound(uint256 roundId) private view returns (bool) {
+    return roundId > 0 && roundId <= type(uint80).max && s_feedState.latestRoundId >= roundId;
+  }
+
+  /// @return L1 sender address
+  function l1Sender() public view virtual returns (address) {
+    return s_l1Sender;
+  }
+
+  /// @notice Set the allowed L1 sender for this contract to a new L1 sender
+  /// @dev Can be disabled by setting the L1 sender as `address(0)`. Accessible only by owner.
+  /// @param to new L1 sender that will be allowed to call `updateStatus` on this contract
+  function transferL1Sender(address to) external virtual onlyOwner {
+    _setL1Sender(to);
+  }
+
+  /// @notice internal method that stores the L1 sender
+  function _setL1Sender(address to) private {
+    address from = s_l1Sender;
+    if (from != to) {
+      s_l1Sender = to;
+      emit L1SenderTransferred(from, to);
+    }
+  }
+
+  /// @dev Returns an AggregatorV2V3Interface compatible answer from status flag
+  /// @param status The status flag to convert to an aggregator-compatible answer
+  function _getStatusAnswer(bool status) private pure returns (int256) {
+    return status ? int256(1) : int256(0);
+  }
+
+  /// @notice Helper function to record a round and set the latest feed state.
+  /// @param roundId The round ID to record
+  /// @param status Sequencer status
+  /// @param timestamp The L1 block timestamp of status update
+  function _recordRound(uint80 roundId, bool status, uint64 timestamp) private {
+    uint64 updatedAt = uint64(block.timestamp);
```

Comment:
I believe it's always cheaper to use block.timestamp (and other calls like it) directly than to assign to a var and then use the var. Not 100% if that's also the case when casting but I'd imagine it would be

Reply:
Same goes for a few more instances in this contract